addons/drag-drop/userscript.js

```diff
@@ -43,4 +56,80 @@ export default async function ({ addon, global, console }) {
     'div[class*="selector_wrapper"]',
     'div[class*="selector_wrapper"] input[class*="action-menu_file-input"]'
   );
+
+  async function listMonitorsDroppable() {
+    while (true) {
+      const listMonitor = await addon.tab.waitForElement('div[class*="monitor_list-monitor"]', { markAsSeen: true });
+      // Get the monitor's context menu ID from the list name
+      // https://github.com/LLK/scratch-gui/blob/develop/src/components/monitor/monitor.jsx#L37
+      const contextMenuId = `monitor-${listMonitor.querySelector('div[class*="monitor_list-header"]').textContent}`;
+      droppable(listMonitor, async (files) => {
+        // Force react-contextmenu's context menu to open using their global
+        // events:
+        // https://github.com/vkbansal/react-contextmenu/blob/v2.9.4/src/ContextMenuTrigger.js#L114-L119
+        // https://github.com/vkbansal/react-contextmenu/blob/v2.9.4/src/actions.js#L27-L29
+        window.dispatchEvent(
+          new CustomEvent("REACT_CONTEXTMENU_SHOW", {
```

Comment:
This looks like a fine way to show the menu, since it matches react-contextmenu's public `actions.js` code, like you linked. I'm curious if simulating a plain right-click event would also work, though? If so, you'd be effectively using the exact same interface as the user themselves do, making the code more reliable in the long-term, avoiding breakage if e.g. the specifics of this CustomEvent (i.e. its event name or expected `detail` format) change.